### DISCUSSION TITLE
consumerMessageBatchMaxSize未生效

### DISCUSSION THREAD

**Discussion by koala775** [apache/rocketmq#8189] [Q&A]
版本：4.4.0
每次消费的条数都是1，消息有堆积，还是1

**Comment by zhiliatom** [comment]
`consumerMessageBatchMaxSize`默认值就是 1，可以在消费者客户端中修改`consumer.setConsumerMessageBatchMaxSize(32)`生效。注意这个值并不可以无限大，因为客户端每次从 Broker 拉取消息到客户端缓存的消息大小、消息条数也是有限制的。服务端通过配置`maxTransferBytesOnMessageInMemory`,`maxTransferCountOnMessageInMemory`,`maxTransferBytesOnMessageInDisk`,`maxTransferBytesOnMessageInDisk`进行限制。
<img width="872" alt="image" src="https://github.com/apache/rocketmq/assets/87265072/aa8f2285-cd84-4555-b40b-9c853702df26">
